ChopChop/Storage/AppDatabase.swift

```diff
@@ -329,45 +355,89 @@ extension AppDatabase {
 extension AppDatabase {
     func saveRecipe(_ recipe: inout RecipeRecord) throws {
         var ingredients: [RecipeIngredientRecord] = []
+        var graph = RecipeStepGraphRecord()
         var steps: [RecipeStepRecord] = []
-
-        try saveRecipe(&recipe, ingredients: &ingredients, steps: &steps)
+        var edges: [RecipeStepEdgeRecord] = []
+        var endPoints: [(source: RecipeStepRecord, destination: RecipeStepRecord)] = []
+
+        try saveRecipe(
+            &recipe,
+            ingredients: &ingredients,
+            graph: &graph,
+            steps: &steps,
+            edges: &edges,
+            endPoints: endPoints)
     }
 
     func saveRecipe(_ recipe: inout RecipeRecord,
                     ingredients: inout [RecipeIngredientRecord],
-                    steps: inout [RecipeStepRecord]) throws {
+                    graph: inout RecipeStepGraphRecord,
+                    steps: inout [RecipeStepRecord],
+                    edges: inout [RecipeStepEdgeRecord],
```

Comment:
This looks like just an empty array, so I'm not sure what its purpose is.

Reply:
I wasn't sure how to test if the edges were added properly because the `RecipeStepEdgeRecord`s are created inside and remain inside the function. Assigning the created edge records to the inout `edges` argument allows testing [like this for example](https://github.com/cs3217-chopchop/ChopChop/blob/3d6d04f0eed7455d4e5902a0ac1f64ad93359f1e/ChopChopTests/Storage/AppDatabaseTests.swift#L315).

I think having an extra argument just for testing is bad for sure but I don't know how to test otherwise